### PR TITLE
HHH-15336 Deprecate SharedSessionContractImplementor#getTransactionStartTimestamp() and CacheTransactionSynchronization#getCurrentTransactionStartTimestamp()

### PR DESCRIPTION
**Pull request by dreab8**
https://hibernate.atlassian.net/browse/HHH-15536

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/engine/spi/SharedSessionContractImplementor.java [hunk 1]
```diff
@@ -167,7 +167,10 @@ default void checkOpen() {
 	 * @apiNote This "timestamp" need not be related to timestamp in the Java Date/millisecond
 	 * sense.  It just needs to be an incrementing value.  See
 	 * {@link CacheTransactionSynchronization#getCurrentTransactionStartTimestamp()}
+	 *
+	 * @deprecated no longer supported, when 2LC is enabled {{@link CacheTransactionSynchronization#getCachingTimestamp()}} can be used.
```
Comment (gavinking) [inline hibernate-core/src/main/java/org/hibernate/engine/spi/SharedSessionContractImplementor.java:171]: It took me a while to figure out that "2LC" meant second-level cache. Could we use spell it out here please?

**Comment by Sanne** [comment]
Do you think it's safe to be backported? I didn't expect that

**Comment by dreab8** [comment]
Hi @Sanne, 
just deprecated the methods we removed in 6.2 and added the new one, why do you think it can be unsafe?

**Review by beikov — APPROVED** [review]
LGTM

**Comment by Sanne** [comment]
> Hi @Sanne, just deprecated the methods we removed in 6.2 and added the new one, why do you think it can be unsafe?

ah, ok then :+1:  Sorry I assumed it was a backport.

**Pull request merged**